samples/ControlCatalog/Pages/ColorPickerPage.xaml

```diff
@@ -15,11 +16,16 @@
 
   <Grid ColumnDefinitions="Auto,10,Auto,10,Auto"
         RowDefinitions="Auto,Auto">
-    <ColorPicker Grid.Column="0"
-                 Grid.Row="1" />
     <ColorView Grid.Column="0"
                Grid.Row="0"
                ColorSpectrumShape="Ring" />
+    <ColorPicker Grid.Column="0"
+                 Grid.Row="1"
+                 Margin="0,50,0,0">
+      <ColorPicker.Palette>
+        <controls:FlatColorPalette />
+      </ColorPicker.Palette>
```

Comment:
Interestingly, this doesn't work.

 * It works if I try to set the palette in XAML on the `ColorView` but does not work on the `ColorPicker`.
 * Other properties work just fine if they are modified as attributes

This may be a larger issue and I'm not sure how to approach it. Setting breakpoints in the property changed method shows `FlatColorPalette` is never set.

Reply:
I wonder if copying static property definition to the ColorPicker class and adding AddOwner fixes this.
Could be part of this larger issue https://github.com/AvaloniaUI/Avalonia/pull/8226

Reply:
I figured out the problem. The default ColorView style sets the `FluentPalette` here:

https://github.com/AvaloniaUI/Avalonia/blob/de28c5ec933388171a0c6a9cffa047fe4208aa68/src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorView.xaml#L87-L89

In a ColorPicker, the ColorView is hosted within a Flyout which won't materialize it's content until it is first opened. So setting this property on ColorPicker at first doesn't do anything. Then why the ColorView opens... it overwrites what the ColorPicker sets.

This doesn't seem correct but also seems extremely difficult to fix.

Reply:
I'm leaving this issue open. The PR can be merged regardless. This is a larger topic I think.